### PR TITLE
clippy: Fix warnings in `components/net`

### PR DESCRIPTION
**Pull request by eerii**
Continuation to #31500, fixes all clippy warnings in `components/net` (excluding safety comments).

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are part of #31500
- [x] These changes do not require tests because they do not modify functionality, they only fix lint errors.

### CODE REVIEW

File: components/net/fetch/methods.rs [hunk 1]
```diff
@@ -873,13 +870,13 @@ async fn scheme_fetch(
...
             StatusCode::SWITCHING_PROTOCOLS |
-            StatusCode::NO_CONTENT |
-            StatusCode::RESET_CONTENT |
-            StatusCode::NOT_MODIFIED => true,
-            _ => false,
-        },
+                StatusCode::NO_CONTENT |
+                StatusCode::RESET_CONTENT |
+                StatusCode::NOT_MODIFIED
+        ),
```
Comment (mrobinson) [inline components/net/fetch/methods.rs:879]: Could this just be something like this:

```
fn is_null_body_status(...) -> bool {
    matches!(
        status,
        Some(StatusCode::SWITCHING_PROTOCOLS) |
        Some(StatusCode::NO_CONTENT) |
        Some(StatusCode::RESET_CONTENT) |
        Some(StatusCode::NOT_MODIFIED)
    )
}

File: components/net/filemanager_thread.rs [hunk 2]
```diff
@@ -824,21 +826,18 @@ impl FileManagerStore {
...
+            self.insert(
+                id,
+                FileStoreEntry {
+                    origin,
+                    file_impl: FileImpl::Memory(blob_buf),
+                    refs: AtomicUsize::new(1),
+                    is_valid_url: AtomicBool::new(set_valid),
+                    outstanding_tokens: Default::default(),
+                },
+            );
```
Comment (mrobinson) [inline components/net/filemanager_thread.rs:840]: Let's make this an early return.

**Review by mrobinson — CHANGES_REQUESTED** [review]
Nice! Just a couple small comments:

File: components/net/filemanager_thread.rs (hunk 2 above)
Comment (eerii) [reply components/net/filemanager_thread.rs]: Done, thanks for the suggestion!

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
Opened #31638 for this intermittent. Sending back to the MQ.

**Pull request merged**